From algernon@balabit.hu Fri Apr 20 12:06:44 2012
Date: Fri, 20 Apr 2012 18:06:31 +0200
From: Gergely Nagy <algernon@balabit.hu>
Reply-To: Syslog-ng users' and developers' mailing list <syslog-ng@lists.balabit.hu>
To: syslog-ng@lists.balabit.hu
Subject: [syslog-ng] [PATCH] afsocket: Fix unix-dgram initialisation

When initializing an unix-dgram socket, set self->fd to the acquired
socket FD, so that fds inherited from systemd actually get used and
polled on.

Signed-off-by: Gergely Nagy <algernon@balabit.hu>
---
 modules/afsocket/afsocket.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/modules/afsocket/afsocket.c b/modules/afsocket/afsocket.c
index 5b192f4..c1bee5c 100644
--- a/modules/afsocket/afsocket.c
+++ b/modules/afsocket/afsocket.c
@@ -679,7 +679,7 @@ afsocket_sd_init(LogPipe *s)
           if (sock == -1 && !afsocket_open_socket(self->bind_addr, !!(self->flags & AFSOCKET_STREAM), &sock))
             return self->super.super.optional;
         }
-      self->fd = -1;
+      self->fd = sock;
 
       if (!self->setup_socket(self, sock))
         {
-- 
1.7.9.1


______________________________________________________________________________
Member info: https://lists.balabit.hu/mailman/listinfo/syslog-ng
Documentation: http://www.balabit.com/support/documentation/?product=syslog-ng
FAQ: http://www.balabit.com/wiki/syslog-ng-faq

